scripts-available/CDB_Helper.sql

```diff
@@ -158,3 +158,21 @@ BEGIN
   RETURN left(string, (i - 1));
 END;
 $$ LANGUAGE 'plpgsql' IMMUTABLE PARALLEL SAFE;
+
+
+-- Checks if a given text representing a qualified or unqualified table name (relation)
+-- actually exists in the database. It is meant to be used as a guard for other function/queries.
+CREATE OR REPLACE FUNCTION cartodb._CDB_Table_Exists(table_name_with_optional_schema TEXT)
+RETURNS bool
+AS $$
+BEGIN
+    IF EXISTS(SELECT * FROM pg_class WHERE table_name_with_optional_schema::regclass::oid = oid AND relkind = 'r') THEN
+       RETURN true;
+    ELSE
+       RETURN false;
+    END IF;
+EXCEPTION
+    WHEN invalid_schema_name OR undefined_table THEN
+        RETURN false;
+END;
+$$ LANGUAGE PLPGSQL VOLATILE PARALLEL SAFE;
```

Comment:
If the function catches exceptions it should usually be declared a PARALLEL UNSAFE [[ref](https://www.postgresql.org/docs/10/static/parallel-safety.html)]:
`Functions and aggregates must be marked PARALLEL UNSAFE if they write to the database, access sequences, change the transaction state even temporarily (e.g. a PL/pgSQL function which establishes an EXCEPTION block to catch errors),`